test/integration/integration_test.go

```diff
@@ -3,244 +3,106 @@
-		appRouteInternal.Spec.Host: internalService.Status.LoadBalancer.Ingress[0].IP,
-	} {
-		err := wait.Poll(1*time.Second, 2*time.Minute, func() (bool, error) {
-			req, err := http.NewRequest("GET", "http://"+ingressIP, nil)
-			req.Host = routeHost
-			resp, err := client.Do(req)
-			defer resp.Body.Close()
-			if err != nil {
-				return false, err
-			}
-			if resp.StatusCode == http.StatusOK {
-				return true, nil
-			}
-			return false, fmt.Errorf("last response: %s", resp.Status)
-		})
-		if err != nil {
-			t.Fatalf("timed out waiting for route endpoint %q at ingress IP %q: %s", routeHost, ingressIP, err)
-		}
-	}
+	// Execute subtests
+	t.Run("TestDefaultIngress", func(t *testing.T) { testDefaultIngress(t, tc) })
 }
 
 type TestConfig struct {
 	operatorNamespace string
 	clusterName       string
+	manifestsDir      string
 	kubeConfig        string
-
-	t *testing.T
 }
 
-func NewTestConfig(t *testing.T) *TestConfig {
-	config := &TestConfig{t: t}
-
+func NewTestConfig(t *testing.T, clusterName string, manifestsDir string) *TestConfig {
 	// Check prerequisites
 	kubeConfig := os.Getenv("KUBECONFIG")
 	if len(kubeConfig) == 0 {
 		t.Fatalf("KUBECONFIG is required")
 	}
 	// The operator-sdk uses KUBERNETES_CONFIG...
 	os.Setenv("KUBERNETES_CONFIG", kubeConfig)
-	config.kubeConfig = kubeConfig
 
-	clusterName := os.Getenv("CLUSTER_NAME")
 	if len(clusterName) == 0 {
-		t.Fatalf("CLUSTER_NAME is required")
+		t.Fatalf("cluster name is required")
+	}
+
+	if len(manifestsDir) == 0 {
+		t.Fatalf("manifests directory is required")
 	}
-	config.clusterName = clusterName
 
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		namespace = "default"
-		os.Setenv("WATCH_NAMESPACE", namespace)
+	return &TestConfig{
+		clusterName:       clusterName,
+		kubeConfig:        kubeConfig,
+		manifestsDir:      manifestsDir,
+		operatorNamespace: "openshift-cluster-ingress-operator",
 	}
-	config.operatorNamespace = namespace
+}
+
+func (tc *TestConfig) setup(t *testing.T) {
+	// uninstall the CVO
+	tc.runShellCmdNonFatal(t, `oc patch -n openshift-cluster-version daemonsets/cluster-version-operator --patch '{"spec": {"template": {"spec": {"nodeSelector": {"node-role.kubernetes.io/fake": ""}}}}}'`)
+
+	// uninstall tectonic-ingress
+	tc.runShellCmdNonFatal(t, `oc delete namespaces/openshift-ingress`)
+
+	// uninstall any existing operator
+	tc.uninstallOperator(t)
 
-	return config
+	// reinstall the operator
+	tc.runShellCmdNonFatal(t, fmt.Sprintf(`oc apply -f %s`, tc.manifestsDir))
 }
 
-func (tc *TestConfig) startOperator() {
-	resource := "ingress.openshift.io/v1alpha1"
-	kind := "ClusterIngress"
-	resyncPeriod := 10 * time.Minute
-	tc.t.Logf("Watching %s, %s, %s, %d", resource, kind, tc.operatorNamespace, resyncPeriod)
-	sdk.Watch(resource, kind, tc.operatorNamespace, resyncPeriod)
-	sdk.Handle(stub.NewHandler())
-	go sdk.Run(context.TODO())
+func (tc *TestConfig) uninstallOperator(t *testing.T) {
```

Comment:
Should all this teardown stuff go in a script so it can be called outside the code? That's what I do for manual testing anyway